### DISCUSSION TITLE
User roles issue in Course Configuration/Email function

### DISCUSSION THREAD

**Discussion by CPA-wrk** [openwebwork/webwork2#2602] [Ideas]
Our department organizes our core math courses with course coordinators, who oversee course content and manage the various instructors for all the sections of their course. We have implemented a WebWork user role of "coordinator" with permission level 11 (something between "professor" and "admin") so that these coordinators have professor-level access to the WebWork courses they oversee, but are not included in emails that students generate using the Email Instructor button. So far so good.

The minor issue I've encountered is that, if a coordinator in fact wants to be included in such emails, say from a particular course/section, there is no option to include a custom user role in the Course Configuration option for the receive_feedback permission level. It appears to me that the list of permission levels for this Email option in Course Configuration is hard coded in ConfigObject/permission_checkboxlist.pm for the default set of roles, rather than using whatever set of levels is currently defined for user roles.

My suggestion is to modify permission_checkboxlist.pm to allow for custom user roles. BTW I use this github account for various projects in addition to my teaching, so it may not be familiar. My WW activity is in connection with my work in the Math department at UMass Boston.

**Comment by Alex-Jordan** [comment]
If I understand right, this issue is already addressed (at least in version 2.19) using a Course Config option in the Email tab. You can select which subset of roles will receive Email Instructor emails. It is a subset, and for example can be set to `ta` and `professor` without including `admin`. Even when the admin email address is present.

**Comment by somiaj** [comment]
@Alex-Jordan It is still a check box list of roles. I understand it as custom roles are not appearing in that list, so they cannot select their custom 'coordinator' role as one of the options.

**Comment by Alex-Jordan** [comment]
Oh, I see. I did misunderstand the post.

It seems that the checkboxes could be generated using keys from `%userRoles`, ordered by their values. But excluding the `nobody` user role. My guess is that hard coding was a cheap way to exclude `nobody` and put these things in the logical order.

**Comment by dlglin** [comment]
Proposed fix in #2603
